lmsweb/views.py

```diff
@@ -147,31 +154,79 @@ def exercises_page():
     )
 
 
+def _create_comment(session_id: int, solution: Solution):
+    user = User.get_or_none(User.id == session_id)
+    if user is None:
+        # should never happen, we checked session_id == solver_id
+        return fail(404, 'No such user')
+
+    kind = request.form.get('kind', '')
+    if (not kind) or (kind not in ('id', 'text')):
+        return fail(400, "Invalid kind")
+
+    try:
+        line_number = int(request.form.get('line', 0))
```

Comment:
Pass as a parameter (this function should not know about the form or how the values are fetched)